src/orchard/namespace.clj

```diff
@@ -20,13 +19,21 @@
 (defn read-namespace
   "Returns the namespace name from the first top-level `ns` form in the file."
   [url]
-  (try
-    (with-open [r (PushbackReader. (io/reader url))]
-      (->> (repeatedly #(edn/read r))
```

Comment:
`repeatedly` + `filter` + clojure's chunked sequences would mean that we would perform more reads than necessary, which is detrimental to perf.